limitador/src/storage/in_memory.rs

```diff
@@ -123,56 +125,44 @@ impl CounterStorage for InMemoryStorage {
                 None
             };
 
-        for counter in counters.iter_mut() {
-            if counter.max_value() < delta {
-                if let Some(limited) = process_counter(counter, 0, delta) {
-                    if !load_counters {
-                        return Ok(limited);
-                    }
-                }
-                continue;
-            }
+        // Normalize counters and values
+        for counter in counters.iter() {
+            limits_by_namespace
+                .entry(counter.limit().namespace().clone())
+                .or_insert_with(HashMap::new)
+                .entry(counter.limit().clone())
+                .or_insert_with(HashMap::new)
+                .entry(counter.into())
+                .or_insert_with(|| {
+                    AtomicExpiringValue::new(0, now + Duration::from_secs(counter.seconds()))
+                });
+        }
```

Comment:
It'd be great to see if we can trigger this on loading a (newer) Limit file. So that this structure is always in sync with the Limits held in memory. This _would need to be atomic_ with the loading of the newer Limits

Reply:
And the value should be `AtomicExpiringValue::default()`... so that the value present _is expired_ already. Otherwise the clock towards expiry would start ticking when the limits are loaded, while the first request might come 30 seconds later... Not a big deal, but it'd change the current behavior.